src/modules/functions.js

```diff
@@ -323,6 +323,192 @@ export const displayUserNameOverlay = () => {
+          let text = node.textContent.trim();
+          if (text === "₣" && nodeList[index + 1] && nodeList[index + 1].nodeType === Node.TEXT_NODE) {
+            const nextText = nodeList[index + 1].textContent.trim();
+            if (!isNaN(parseFloat(nextText))) {
+              const combinedText = text + nextText;
+              const tokenValue = parseFloat(combinedText.slice(1));
+              if (!isNaN(tokenValue)) {
+                const localCurrencyValue = (tokenValue * tokenToUsdRate * usdExchangeRate).toFixed(2);
+                node.textContent = `$${localCurrencyValue}`;
+                nodeList[index + 1].textContent = '';
+              }
+            }
+          } else if (text.startsWith("₣")) {
+            const tokenValue = parseFloat(text.slice(1));
+            if (!isNaN(tokenValue)) {
+              const localCurrencyValue = (tokenValue * tokenToUsdRate * usdExchangeRate).toFixed(2);
+              node.textContent = `$${localCurrencyValue}`;
+            }
+          }
+        }
+      });
+    } else {
+      const originalText = element.textContent.trim();
+      const tokenMatch = originalText.match(/₣(\d+(\.\d+)?)/);
+
+      if (tokenMatch) {
+        const tokenValue = parseFloat(tokenMatch[1]);
+        if (!isNaN(tokenValue)) {
+          const localCurrencyValue = (tokenValue * tokenToUsdRate * usdExchangeRate).toFixed(2);
+          element.innerHTML = originalText.replace(tokenMatch[0], `$${localCurrencyValue}`);
+        }
+      }
+    }
+
+    if (element.classList.contains(ELEMENTS.token.ttsModalTokens.class) || element.classList.contains(ELEMENTS.token.sfxModalTokens.class)) {
+      element.style.width = "135px";
+    }
+  };
+
+  const revertToOriginalTokens = (element) => {
+    const originalContent = element.getAttribute("data-original");
+    if (originalContent) {
+      element.innerHTML = originalContent;
+      element.removeAttribute("data-original");
+    }
+    if (element.classList.contains(ELEMENTS.token.ttsModalTokens.class) || element.classList.contains(ELEMENTS.token.sfxModalTokens.class)) {
+      element.style.width = "96px";
+    }
+  };
+
+  const processElements = () => {
+    const selectors = [
+      ELEMENTS.token.topBarUserTokens.selector,
+      ELEMENTS.token.ttsModalTokens.selector,
+      ELEMENTS.token.sfxModalTokens.selector,
+      ELEMENTS.token.toysFishtoysTokens.selector,
+      ELEMENTS.token.buyTokensModal.selector,
+      ELEMENTS.token.voteModalTokens.selector + " span"
+    ];
+
+    selectors.forEach((selector) => {
+      document.querySelectorAll(selector).forEach((element) => {
+        if (element.closest(ELEMENTS.token.toysBigToyPrice.selector)) return;
+        if (toggle) {
+          convertTokensToLocalCurrency(element);
+        } else {
+          revertToOriginalTokens(element);
+        }
+      });
+    });
+  };
+
+  // Initial processing
+  processElements();
+};
+
+export const togglePopoutChatButton = (toggle) => {
```

Comment:
left a [comment](https://github.com/f3rked/maej3rked-tools/pull/29#pullrequestreview-2435151049) about removing this

Reply:
Ive changed this to just add a button at the top right of the chat element that opens a link to fishtank.live/chat is that okay or you want that removed too?

Reply:
my bad - probably should have waited longer after i woke up to review.

that sounds good.  i'll test later today to see what it looks like.